src/main/java/com/algorand/algosdk/v2/client/indexer/LookupAccountTransactions.java

```diff
@@ -0,0 +1,260 @@
+		return this;
+	}
+
+	/**
+	 * Maximum number of results to return. 
+	 */
+	public LookupAccountTransactions setLimit(long limit) {
+		this.limit = limit;
+		this.limitIsSet = true;
+		return this;
+	}
+
+	/**
+	 * Include results at or before the specified max-round. 
+	 */
+	public LookupAccountTransactions setMaxRound(long maxRound) {
+		this.maxRound = maxRound;
+		this.maxRoundIsSet = true;
+		return this;
+	}
+
+	/**
+	 * Include results at or after the specified min-round. 
+	 */
+	public LookupAccountTransactions setMinRound(long minRound) {
+		this.minRound = minRound;
+		this.minRoundIsSet = true;
+		return this;
+	}
+
+	/**
+	 * The next page of results. Use the next token provided by the previous results. 
+	 */
+	public LookupAccountTransactions setNext(String next) {
+		this.next = next;
+		this.nextIsSet = true;
+		return this;
+	}
+
+	/**
+	 * Specifies a prefix which must be contained in the note field. 
+	 */
+	public LookupAccountTransactions setNotePrefix(String notePrefix) {
+		this.notePrefix = notePrefix;
+		this.notePrefixIsSet = true;
+		return this;
+	}
+
+	/**
+	 * Include results for the specified round. 
+	 */
+	public LookupAccountTransactions setRound(long round) {
+		this.round = round;
+		this.roundIsSet = true;
+		return this;
+	}
+
+	/**
+	 * SigType filters just results using the specified type of signature: * sig - 
+	 * Standard * msig - MultiSig * lsig - LogicSig 
+	 */
+	public LookupAccountTransactions setSigType(String sigType) {
+		this.sigType = sigType;
+		this.sigTypeIsSet = true;
+		return this;
+	}
+
+	/**
+	 * Lookup the specific transaction by ID. 
+	 */
+	public LookupAccountTransactions setTxId(String txId) {
+		this.txId = txId;
+		this.txIdIsSet = true;
+		return this;
+	}
+	public LookupAccountTransactions setTxType(String txType) {
+		this.txType = txType;
+		this.txTypeIsSet = true;
+		return this;
+	}
+
+	public TransactionsResponse lookup() {
+		String response;
+		try {
+			response = request("get");
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			return null;
+		}
+		ObjectMapper mapper = new ObjectMapper();
+		TransactionsResponse resp;
+		try {
+			resp = mapper.readValue(response, TransactionsResponse.class);
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			return null;
+		}
+		return resp;
+	}
+	public QueryData getRequestString() {
+		QueryData qd = new QueryData();
+		if  (!this.accountIdIsSet) {
+			throw new RuntimeException("accountId is not set, and it is a required parameter.");
+		}
+		if (this.afterTimeIsSet) {
+			qd.addQuery("afterTime", String.valueOf(afterTime));
+		}
+		if (this.assetIdIsSet) {
+			qd.addQuery("assetId", String.valueOf(assetId));
+		}
+		if (this.beforeTimeIsSet) {
+			qd.addQuery("beforeTime", String.valueOf(beforeTime));
+		}
+		if (this.currencyGreaterThanIsSet) {
+			qd.addQuery("currencyGreaterThan", String.valueOf(currencyGreaterThan));
```

Comment:
Instead of having this big section of if statements, you could add a `QueryData` object to the parent object, and `addParemeter` helper methods. Then you could just call `addQuery("currencyGreaterThan", currencyGreaterThan)` with no conditional.

You could also have multiple `addQuery` overloads to support different types.

Reply:
Fixed.